samtranslator/model/sam_resources.py

```diff
@@ -2144,26 +2147,76 @@ class SamGraphQLApi(SamResourceMacro):
     Name: Optional[str]
     SchemaInline: Optional[str]
     SchemaUri: Optional[str]
+    Logging: Optional[Union[Dict[str, Any], bool]]
 
     @cw_timer
     def to_cloudformation(self, **kwargs: Any) -> List[Resource]:
-        appsync_api = self._construct_appsync_api()
+        appsync_api, cloudwatch_role = self._construct_appsync_api_resources()
         appsync_schema = self._construct_appsync_schema(appsync_api.get_runtime_attr("api_id"))
+
         resources: List[Resource] = [appsync_api, appsync_schema]
 
+        if cloudwatch_role:
+            resources.append(cloudwatch_role)
+
         return resources
 
-    def _construct_appsync_api(self) -> GraphQLApi:
+    def _construct_appsync_api_resources(self) -> Tuple[GraphQLApi, Optional[IAMRole]]:
         api = GraphQLApi(logical_id=self.logical_id, depends_on=self.depends_on, attributes=self.resource_attributes)
 
         api.AuthenticationType = sam_expect(self.Auth.get("Type"), self.logical_id, "Auth.Type").to_be_a_string()
         api.Name = self.Name if self.Name else self.logical_id
         api.XrayEnabled = self.XrayEnabled
 
+        cloudwatch_role = None  # conditionally generated, if left as None then its not added to resources
+
+        # Logging has 3 possible types: dict, bool, and None.
+        # GraphQLApi will not include logging if and only if the user explicity sets Logging as false boolean.
+        # It will for every other value (including true boolean which is essentially same as None).
+        if type(self.Logging) != bool or self.Logging is True:
```

Comment:
It’s hard to pull off a simple negation for this since there’s two cases where we don’t need to create a role, Logging disabled, or if customer already has a Role. But in the case that customer already has a role, we still need to create the LogConfig property, so we’ll have some additional if statements.

I’m working on a change that would extract the role creation logic to the `to_cloudformation()` function, so then each function will just return a single resource and I can get rid of these conditional tuple returns. It should be a bit more clear as to when and why a CloudWatch role is created for the customer.